great_expectations/experimental/datasources/sql_datasource.py

```diff
@@ -827,7 +823,12 @@ class SQLDatasource(Datasource):
 
     # private attrs
     _cached_connection_string: str = pydantic.PrivateAttr("")
-    _engine: Union[sqlalchemy.engine.Engine, None] = pydantic.PrivateAttr(None)
+    # The sqlalchemy is in quotes (hence the # noqa) here because otherwise pydantic
+    # will try to resolve this annotation at gx import time which will cause the
+    # process die if sqlalchemy is not installed.
+    _engine: Union["sqlalchemy.engine.Engine", None] = pydantic.PrivateAttr(  # noqa
```

Comment:
Please see comment above. The quotes aren't for static analysis and type checking, they are there because of pydantic. I am not sure what the correct code to use is for the `noqa` annotation. Is there a definitive ruff list somewhere?

Reply:
Oh I see it now, it's `UP037`. I'll update it.